### DISCUSSION TITLE
Ability to turn watch off or put into deep sleep

### DISCUSSION THREAD

**Discussion by riban-bw** [InfiniTimeOrg/InfiniTime#1982] [Ideas]
The watch remains on at all times. Its display sleeps and you can configure it to only _wake_ under certain conditions, e.g. button press, shake, etc. It would be advantageous to be able to turn the watch off completely or at least put it into a deep sleep mode, e.g. when you are away from any charging points for prolonged periods and need to peserve battery or when you need the device completely inactive due to environmental conditions, e.g. security demands of a building.

Currently a long press of the button will restart the watch. It would be good to offer an option to restart / turn off the watch. In the _OFF_ state the watch would respond to a long button press to turn it on. I appreciate that we may need to keep the watch powered (depending on the hardware configuration of the button and power management) but at the very least, putting it into a deep sleep with all functionality disabled except the wake-on-long-button-press would be a reasonable solution.

**Comment by JF002** [comment]
When InfiniTime turns the display off, it also set the MCU (and all other devices like the sensors, spi memory,...) to enter in deep sleep mode to reduce power usage as much as possible. I think the NRF52 has deeper sleep modes, but they require a full reset of the MCU to wake up, which would also erase the content of the RAM memory, and that would not be very convenient.

It *might* be possible to manually enter into one of these deeper sleep mode, and use the button to generate a reset to wake up the watch, but in any cases, the PineTime does not have any hardware that allows to cut the power to the MCU and other devices.
